examples/gno.land/p/demo/ownable/ownable.gno

```diff
@@ -4,23 +4,42 @@ import (
 	"std"
 )
 
-// Ownable is meant to be used as a top-level object to make your contract ownable OR
-// being embedded in a Gno object to manage per-object ownership.
-type Ownable struct {
+// Ownable defines an interface that requires a method to get the owner's address
+type Ownable interface {
+	Owner() std.Address
+}
+
+// Transferrable embeds Ownable, indicating that an object can transfer and drop an ownership
+type Transferrable interface {
+	Ownable
+
+	TransferOwnership(newOwner std.Address) error
+	DropOwnership() error
+}
+
+// TransferrableOwnership provides a basic implementation of ownership functionality
+// It can be used as a top-level object to make a contract ownable or embedded within
+// another object to manage per-object ownership.
+type TransferrableOwnership struct {
```

Comment:
You can unexport it so people will only use the interfaces and the constructors. And you can simply name it transferable, as transferable embeds ownable.

```suggestion
type transferrable struct {
```

Reply:
it's make sense